crates/turborepo-api-client/src/lib.rs

```diff
@@ -100,6 +100,18 @@ pub struct APIAuth {
     pub team_slug: Option<String>,
 }
 
+impl APIAuth {
+    pub fn is_linked(&self) -> bool {
+        !self.token.is_empty()
```

Comment:
Want to defer to the auth crate here? cc @Zertsov

Reply:
Yes, but I think it needs some refactoring to happen. `APIAuth` might properly be owned by that crate, although we'll have to work out how that interacts with the config code.

Reply:
I don't think this is correct. If there is no token or team id, `APIAuth` [shouldn't even be constructed](https://github.com/vercel/turbo/blob/ef2b800167d527f3fa8bc06ac98f7edaeb51b410/crates/turborepo-lib/src/run/mod.rs#L170). To check if linked we should just check if `Option<APIAuth>` is `Some`

Reply:
Ah ok. That makes sense, if having an `APIAuth` implies that we at least have a token and a team id, then that works

Reply:
Switched to `is_linked = api_auth.is_some()`